pinot-common/src/main/java/org/apache/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -103,10 +109,26 @@ public BrokerRequest compileToBrokerRequest(String expression)
       validateHavingClause(rootNode);
 
       BrokerRequest brokerRequest = new BrokerRequest();
-      PinotQuery pinotQuery = new PinotQuery();
-      brokerRequest.setPinotQuery(pinotQuery);
       rootNode.updateBrokerRequest(brokerRequest);
-      rootNode.updatePinotQuery(pinotQuery);
+      try {
+        PinotQuery pinotQuery = new PinotQuery();
+        rootNode.updatePinotQuery(pinotQuery);
+        boolean validateConverter = true;
+        if (validateConverter) {
+          PinotQuery2BrokerRequestConverter converter = new PinotQuery2BrokerRequestConverter();
+          BrokerRequest tempBrokerRequest = converter.convert(pinotQuery);
+          tempBrokerRequest.setQueryOptions(pinotQuery.getQueryOptions());
```

Comment:
QueryOptions is already set insider converter.convert.

Reply:
will fix that.